examples/player/simid_player.js

```diff
@@ -609,16 +602,78 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when video playback is complete.
+   * @private
+   */
+  videoComplete() {
+    this.simidProtocol.sendMessage(MediaMessage.ENDED);
+
+    if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+      this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+    }
+  }
+
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    const originalDuration = this.requestedDuration_;
+    this.requestedDuration_  = incomingMessage.args['duration'];
+    if (this.requestedDuration_ != UNLIMITED_DURATION && this.requestedDuration_< 0) {
+      const durationErrorMessage = {
+        errorCode: PlayerErrorCode.UNSUPPORTED_TIME,
+        message: 'A negative duration is not valid.'
+      }
+      this.requestedDuration_ = originalDuration;
+      this.simidProtocol.reject(incomingMessage, durationErrorMessage);
+    } else if (this.requestedDuration_ == UNLIMITED_DURATION) {
+      //Note: Users can end the ad using the close ad button on the player
+      return;
+    } else {
+      //If requested duration is any other acceptable value
+      this.changeAdEndTime();
+      this.simidProtocol.resolve(incomingMessage);
     }
-    const requestedDuration  = incomingMessage.args['duration'];
-    this.requestedDuration_ = requestedDuration;
-    this.simidProtocol.resolve(incomingMessage);
   }
 
+  /**
+   * Compares the requested change duration with the current ad time
+   *  to determine when to stop the ad, every 250 milliseconds.
+   * @private
+   */
+  changeAdEndTime() {
+    this.interval_ = setInterval(() => {
+        this.compareAdAndRequestedDurations(); 
+    }, INTERVAL_TIMER);
+  }
+
+  /**
+   * Compares the duration of the ad with the requested change duration.
+   * If request duration is shorter, the ad stops early. 
+   * If the request duration is longer, the ad extends for the requested
+   *  amount of time.
+   * @private
+   */
+  compareAdAndRequestedDurations() {
+    const durationChangeMs = (this.requestedDuration_ - this.adVideoElement_.duration) * 1000;
+
+    if (this.adVideoElement_.ended) {
+      //If the video ad has ended already close the creative at the requested duration
+      setTimeout(() => {
+        this.stopAd(StopCode.CREATIVE_INITIATED);
+      }, durationChangeMs);
+      clearInterval(this.interval_);
+      return;
```

Comment:
Seems odd to use a timer to check if the video element has ended, when it dispatches an ENDED event that we are already listening for.  Can this logic be moved to the videoComplete function?

Reply:
@ryanthompson591 I moved the logic to the videoComplete and it didn't seem to cause any issues. Do you have a strong opinion on keeping this check within this function or do you think it can be moved?

Reply:
spoke with Ryan and he said either is fine, so moved it